### PR TITLE
feat: azure ai search rag

### PR DESCRIPTION
**Pull request by xiaolang124**
Add azure ai search rag template.

Work item: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/27548175/

TODO: update github wiki and readme wiki part.

image

### CODE REVIEW

File: templates/ts/custom-copilot-rag-azure-ai-search/README.md.tpl [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+{{#useOpenAI}}
+> - An account with [OpenAI](https://platform.openai.com/) and [Azure AI Search](https://azure.microsoft.com/en-us/products/ai-services/ai-search).
+{{/useOpenAI}}
+{{#useAzureOpenAI}}
+> - Prepare your own [Azure OpenAI](https://aka.ms/oai/access) resource and [Azure AI Search](https://azure.microsoft.com/en-us/products/ai-services/ai-search).
+{{/useAzureOpenAI}}
+
+1. First, select the Teams Toolkit icon on the left in the VS Code toolbar.
+{{#useOpenAI}}
+1. In file *env/.env.testtool.user*, fill in your OpenAI key `SECRET_OPENAI_API_KEY=<your-key>`. And fill in your Azure AI search key `SECRET_AZURE_SEARCH_KEY=<your--ai-search-key>` and endpoint `AZURE_SEARCH_ENDPOINT=<your-ai-search-endpoint>`.
```
Comment (frankqianms) [inline templates/ts/custom-copilot-rag-azure-ai-search/README.md.tpl:29]: minor:  As there is no <your-key> placeholder in .env.*.user, do we need to use this placeholder here?  @MuyangAmigo What's your opinion?

File: templates/ts/custom-copilot-rag-azure-ai-search/teamsapp.local.yml.tpl [hunk 2]
```diff
@@ -0,0 +1,93 @@
+# yaml-language-server: $schema=https://aka.ms/teams-toolkit/1.0.0/yaml.schema.json
+# Visit https://aka.ms/teamsfx-v5.0-guide for details on this file
+# Visit https://aka.ms/teamsfx-actions for details on actions
+version: 1.0.0
```
Comment (frankqianms) [inline templates/ts/custom-copilot-rag-azure-ai-search/teamsapp.local.yml.tpl:4]: may upgrade it to V1.3

File: templates/ts/custom-copilot-rag-azure-ai-search/teamsapp.yml.tpl [hunk 3]
```diff
@@ -0,0 +1,145 @@
+# yaml-language-server: $schema=https://aka.ms/teams-toolkit/1.0.0/yaml.schema.json
+# Visit https://aka.ms/teamsfx-v5.0-guide for details on this file
+# Visit https://aka.ms/teamsfx-actions for details on actions
+version: 1.0.0
```
Comment (frankqianms) [inline templates/ts/custom-copilot-rag-azure-ai-search/teamsapp.yml.tpl:4]: V1.3, same here

File: templates/ts/custom-copilot-rag-azure-ai-search/env/.env.dev.user.tpl [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+SECRET_AZURE_OPENAI_API_KEY=
+{{/azureOpenAIKey}}
+{{#azureOpenAIEndpoint}}
+AZURE_OPENAI_ENDPOINT='{{{azureOpenAIEndpoint}}}'
+{{/azureOpenAIEndpoint}}
+{{^azureOpenAIEndpoint}}
+AZURE_OPENAI_ENDPOINT=
+{{/azureOpenAIEndpoint}}
+{{#azureOpenAIDeploymentName}}
+AZURE_OPENAI_DEPLOYMENT_NAME='{{{azureOpenAIDeploymentName}}}'
```
Comment (kuojianlu) [inline templates/ts/custom-copilot-rag-azure-ai-search/env/.env.dev.user.tpl:27]: Remember to update the scaffolding question model to support this.

File: templates/ts/custom-copilot-rag-azure-ai-search/package.json.tpl [hunk 5]
```diff
@@ -0,0 +1,46 @@
+{
+    "name": "{{SafeProjectNameLowerCase}}",
+    "version": "1.0.0",
+    "msteams": {
+        "teamsAppId": null
+    },
+    "description": "Microsoft Teams Toolkit AI Chat Bot Sample with Teams AI Library",
```
Comment (kuojianlu) [inline templates/ts/custom-copilot-rag-azure-ai-search/package.json.tpl:7]: Not AI Chat Bot.

File: templates/ts/custom-copilot-rag-azure-ai-search/env/.env.dev.user.tpl [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+{{/azureOpenAIDeploymentName}}
+{{#azureOpenAIEmbeddingDeploymentName}}
+AZURE_OPENAI_EMBEDDING_DEPLOYMENT_NAME='{{{azureOpenAIEmbeddingDeploymentName}}}'
+{{/azureOpenAIEmbeddingDeploymentName}}
+{{^azureOpenAIEmbeddingDeploymentName}}
+AZURE_OPENAI_EMBEDDING_DEPLOYMENT_NAME=
+{{/azureOpenAIEmbeddingDeploymentName}}
+{{/useAzureOpenAI}}
+{{#secretAzureSearchKey}}
+SECRET_AZURE_SEARCH_KEY='{{{secretAzureSearchKey}}}'
```
Comment (kuojianlu) [inline templates/ts/custom-copilot-rag-azure-ai-search/env/.env.dev.user.tpl:40]: Do we need this? I assume the azure search settings will not be asked when scaffolding, right?

File: templates/ts/custom-copilot-rag-azure-ai-search/env/.env.dev.user.tpl (hunk 4 above)
Comment (frankqianms) [reply templates/ts/custom-copilot-rag-azure-ai-search/env/.env.dev.user.tpl:27]: Do we need to ask user these deployment name when scaffolding?

**Review by frankqianms — APPROVED** [review]
some minor things can be fixed in another pr

**Review by kuojianlu — APPROVED** [review]

**Pull request merged**